src/containers/Supplier/Supplier.tsx

```diff
@@ -1,16 +1,93 @@
+import axios from 'axios'
+import { createContext, useContext, useState } from 'react'
+import { QueryParamsType } from 'src/@types/user'
 import AddComp from './Table/Components/AddComp'
 import TableSupplier from './Table/TableSupplier'
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+export interface Supplier {
+  id: string
+  nama: string
+  alamat: string
+  nomor_telepon: string
+}
+
+interface Data {
+  data: Supplier[],
+  total: number;
+}
+
+interface supplierContext {
+  data: Data;
+  setData: React.Dispatch<React.SetStateAction<Data>>;
+  queryParams: QueryParamsType;
+  setQueryParams: React.Dispatch<React.SetStateAction<QueryParamsType>>;
+  refreshSupplier: () => void;
+}
+
+const defaultState = {
+  data: { data: [], total: 0 },
+  setData: () => {},
+  queryParams: {
+    keywords: '',
+    orderBy: 'nama',
+    orderType: 'ASC',
+    page: 1,
+    limit: 10,
+  },
+  setQueryParams: () => {},
+  refreshSupplier: () => {},
+}
+
+const supplierContext = createContext<supplierContext>(defaultState)
+
+export const useSupplierContext = () => {
+  return useContext(supplierContext)
+} 
 
 export const DataSupplier = () => {
+  const [data, setData] = useState<Data>({
+    data: [],
+    total: 0,
+  });
+
+  const [queryParams, setQueryParams] = useState<QueryParamsType>({
+    keywords: '',
+    orderBy: 'nama',
+    orderType: '',
+    page: 1,
+    limit: 10,
+  })
```

Comment:
its defined in defaultState so use it. 
`useState<QueryParamsType>(defaultState.queryParams)`

Reply:
done, thanks for the info